preview-env/create/action.yml

```
@@ -68,10 +68,39 @@ runs:
       ref: ${{ inputs.revision }}
   - name: Deploy
     shell: bash
-    run: >-
-      argocd app create ${{ inputs.argocd_arguments }}
+    run: |
+      max_retries=3
+      for attempt in $(seq "$max_retries"); do
+          argocd app create ${{ inputs.argocd_arguments }}
+          if [ $? -eq 0 ]; then
+            echo "Creating Argo CD app with name ${{ inputs.app_name }}"
+            sleep 60
```

@andromaqui sorry for jumping on it, but why we sleep for a whole minute here? Eventual consistency app state convergence on the Argo side would be much faster than this, I don't see the need to sleep for a whole minute always, especially we have a retry mechanism in place. This makes a pipeline a whole minute longer for in my opinion not a good enough reason, negatively impacting user experience and increasing GHA costs.

I'd recommend something more moderate instead:
```suggestion
            sleep 10
```

---

yeah you are right since we retry 3 times maybe it is not very necessay, I will address this 👍🏻